plugin-cgroupv1/src/oar2/plugin.rs

```diff
@@ -270,6 +134,104 @@ impl AlumetPlugin for Oar2Plugin {
+        log::debug!("CPU path {cpuacct_controller_job_path:?}");
+        let memory_controller_job_path = memory_controller_path.join(job_name);
+        log::debug!("Memory path {memory_controller_job_path:?}");
+
+        let cpuacct_usage_filepath = cpuacct_controller_job_path
+            .join("cpuacct.usage")
+            .to_str()
+            .unwrap()
+            .to_string();
+        let memory_usage_filepath = memory_controller_job_path
+            .join("memory.usage_in_bytes")
+            .to_str()
+            .unwrap()
+            .to_string();
+
+        Ok(Box::new(OAR2Probe::new(
+            job_id,
+            metrics,
+            Some(cpuacct_usage_filepath),
+            Some(memory_usage_filepath),
+        )?))
+    }
+}
+
+struct JobDetector {
+    cpuacct_controller_path: PathBuf,
+    memory_controller_path: PathBuf,
+    control_handle: PluginControlHandle,
+    metrics: Metrics,
+    trigger: TriggerSpec,
+    rt: tokio::runtime::Runtime,
+}
+
+impl EventHandler for JobDetector {
+    fn handle_event(&mut self, event: Result<Event, notify::Error>) {
+        fn handle_event_on_path(job_detect: &mut JobDetector, path: PathBuf) -> anyhow::Result<()> {
+            if let Some(job_name) = path.file_name() {
+                let job_name = job_name.to_str().unwrap().to_string();
+
+                if job_name.chars().any(|c| c.is_numeric()) {
+                    let source = Oar2Plugin::new_job_source(
+                        &job_name,
+                        job_detect.metrics.clone(),
+                        &job_detect.cpuacct_controller_path,
+                        &job_detect.memory_controller_path,
+                    )?;
+
+                    let source_name = job_name;
+                    let new_job_source =
+                        request::create_one().add_source(&source_name, source, job_detect.trigger.clone());
+
+                    job_detect
+                        .rt
+                        .block_on(
+                            job_detect
+                                .control_handle
+                                .dispatch(new_job_source, Duration::from_millis(500)),
+                        )
+                        .with_context(|| format!("failed to add source {source_name}"))?;
+                }
+            }
+            Ok(())
+        }
+
+        log::debug!("Handle event function");
+        if let Ok(Event {
+            kind: EventKind::Create(_),
+            paths,
+            ..
+        }) = event
+        {
+            log::debug!("Paths: {paths:?}");
+            for path in paths {
+                if let Err(e) = handle_event_on_path(self, path.clone()) {
+                    log::error!("Unable to handle event on {}: {}", path.display(), e);
+                }
+            }
+        } else if let Err(e) = event {
+            log::error!("watch error: {:?}", e);
+        }
```

Comment:
Style improvement: use a `match` here to clearly show the different possibilities: Ok with a kind of Create, Ok with other content, and Err.